File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text;
    int countLetters;
    int countWords;
    int countSentences;

    float L;
    float S;
    float index;

    text = get_string("Text: ");

    countLetters = count_letters(text);
    countWords = count_words(text);
    countSentences = count_sentences(text);

    L = ((float)countLetters / countWords) * 100.0;
    S = ((float)countSentences / countWords) * 100.0;
    index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);

    //Checks what to print based on the index value
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", (int)index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}



int count_letters(string text)
{
    int totalLetters = 0;

    //Loop that evaluates each individual character in the string text
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        //If character is lowercase letter or uppercase, increase counter by 1
        if (islower(text[i]))
        {
            totalLetters += 1;
        }
        else if (isupper(text[i]))
        {
            totalLetters += 1;
        }
        //If character is other, then add 0 to the counter
        else
        {
            totalLetters += 0;
        }

    }

    return totalLetters;
}


int count_words(string text)
{
    int totalWords = 0;

    //Loop that evaluates each individual character in the string text
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        //If a space is encountered, then there was a word before it. Increase word counter by 1
        if (text[i] == ' ')
        {
            totalWords += 1;
        }
    }

    //This final addition to the word counter accounts for the last word (no space after final word)
    totalWords += 1;
    return totalWords;
}


int count_sentences(string text)
{
    int totalSentences = 0;

    //Loop that evaluates each individual character in the string text
    for (int i = 0, l = strlen(text); i < l; i++)
    {
        //If a '.', '!', or '?' is found, then the sentence counter increases by 1
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            totalSentences += 1;
        }
    }

    return totalSentences;
}

File: sentimental-readability/readability.py
# python readability.py
from cs50 import get_string


def main():
    # Get String from user
    text = get_string("Text: ")

    # Count the number of letters, words, and sentences
    countLetters = count_letters(text)
    countWords = count_words(text)
    countSentences = count_sentences(text)

    # Compute the Coleman-Liau index
    L = (countLetters / countWords) * 100.0
    S = (countSentences / countWords) * 100.0
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index, 0)

    # Print the Grade (ie. index) accordingly
    if index >= 1 and index < 16:
        print(f"Grade {index}")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")


# Count the number of letters in the text
def count_letters(text):
    totalLetters = 0

    for i in range(0, len(text)):
        # Check if the current character is a lower case letter
        if ord(text[i]) >= 97 and ord(text[i]) <= 122:
            totalLetters += 1
        # Check if the current character is an upper case letter
        elif ord(text[i]) >= 65 and ord(text[i]) <= 90:
            totalLetters += 1
        # Check if the current character is anything but a letter
        else:
            totalLetters += 0

    return totalLetters


# Counts the number of words in the text
def count_words(text):
    totalWords = 0

    for k in range(0, len(text)):
        # Count the number of spaces (a space comes after each word)
        if text[k] == " ":
            totalWords += 1

    # Account for the last word (there is no space after the last word)
    totalWords += 1
    return totalWords


# Counts the number of senteces in the text
def count_sentences(text):
    totalSentences = 0

    for z in range(0, len(text)):
        # Assume that a ., ?, or ! indicates the end of a sentence
        if text[z] == '.' or text[z] == '!' or text[z] == '?':
            totalSentences += 1

    return totalSentences


main()
